### PR TITLE
[APPS-2135] [APPS-2155] migration of dependency from moment to date-fns

### PR DESCRIPTION
**Pull request by jatin2008**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
moment dependency and moment adapter is present in the component

**What is the new behaviour?**
migration of dependency in the component from moment to date-fns

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: docs/eslint-angular/rules/use-none-component-view-encapsulation.md
Comment (AleksanderSklorz) [inline docs/eslint-angular/rules/use-none-component-view-encapsulation.md:1]: Looks like that PR contains changes out of scope of that jira. There was probably some issue during synchronizing your branch with develop so it shows not your changes. Please correct and make visible only your changes.

**Review by AleksanderSklorz — APPROVED** [review]

File: lib/eslint-angular/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Alfresco ESlint Angular Library
```
Comment (DenysVuika) [inline lib/eslint-angular/README.md:1]: this is probably incorrect rebase, as it belongs to another PR

File: docs/eslint-angular/rules/use-none-component-view-encapsulation.md [hunk 2]
```diff
@@ -7,11 +7,12 @@ Last reviewed: 2023-05-23
 
 # [Use none component view encapsulation](../../../lib/eslint-angular/src/rules/use-none-component-view-encapsulation/use-none-component-view-encapsulation.ts "Defined in use-none-component-view-encapsulation.ts")
 
-Custom ESLint rule which check if component uses ViewEncapsulation.None. It has been implemented because None encapsulation makes themes styling easier.
-It also allows to autofix.
+Custom ESLint rule that requires components to use ViewEncapsulation.None. It has been implemented because None encapsulation makes themes styling easier.
```
Comment (DenysVuika) [inline docs/eslint-angular/rules/use-none-component-view-encapsulation.md:10]: This is incorrect rebase as it belongs to another PR

File: lib/content-services/src/lib/search/components/search-date-range/search-date-range.component.ts [hunk 3]
```diff
@@ -234,8 +228,7 @@ export class SearchDateRangeComponent implements SearchWidget, OnInit, OnDestroy
     }
 
     setLocale(locale) {
```
Comment (DenysVuika) [inline lib/content-services/src/lib/search/components/search-date-range/search-date-range.component.ts:230]: please provide the missing typings here, probably "locale: string"

File: lib/eslint-angular/README.md (hunk 1 above)
Comment (jatin2008) [reply lib/eslint-angular/README.md]: rebased the branch - APSS-2135-replace-moment-to-date-fns now only my changes related to moment to date-fns conversion are showing.

File: docs/eslint-angular/rules/use-none-component-view-encapsulation.md (hunk 2 above)
Comment (jatin2008) [reply docs/eslint-angular/rules/use-none-component-view-encapsulation.md]: rebased the branch - APSS-2135-replace-moment-to-date-fns now only my changes related to moment to date-fns conversion are showing.

File: lib/content-services/src/lib/search/components/search-date-range/search-date-range.component.ts (hunk 3 above)
Comment (jatin2008) [reply lib/content-services/src/lib/search/components/search-date-range/search-date-range.component.ts]: added type here

File: lib/content-services/src/lib/search/components/search-date-range/search-date-range.component.ts [hunk 4]
```diff
@@ -234,8 +228,7 @@ export class SearchDateRangeComponent implements SearchWidget, OnInit, OnDestroy
...
-        this.dateAdapter.setLocale(locale);
-        moment.locale(locale);
+        this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale));
     }
 
     hasParseError(formControl): boolean {
```
[the remark this answers is not in the document]
Comment (jatin2008) [reply lib/content-services/src/lib/search/components/search-date-range/search-date-range.component.ts]: added type here

**Review by DenysVuika — APPROVED** [review]
